guardrails/run/runner.py

```diff
@@ -380,7 +397,9 @@ def prepare_msg_history(
 
         # validate msg_history
         if "msg_history" in self.validation_map:
-            self.validate_msg_history(call_log, formatted_msg_history, attempt_number)
+            formatted_msg_history = self.validate_msg_history(  # type: ignore
```

Comment:
Is this msg_history being mutated during validation?

Reply:
deep let us deep copy this